proxy/logging/LogConfig.cc

```diff
@@ -818,36 +836,61 @@ LogConfig::update_space_used()
   if (candidate_count > 0 && !space_to_write(headroom)) {
     Debug("logspace", "headroom reached, trying to clear space ...");
     Debug("logspace", "sorting %d delete candidates ...", candidate_count);
-    qsort(candidates, candidate_count, sizeof(LogDeleteCandidate), (int (*)(const void *, const void *))delete_candidate_compare);
 
-    for (victim = 0; victim < candidate_count; victim++) {
+    deleting_info.apply([](LogDeletingInfo &info) {
+      std::sort(info.candidates.begin(), info.candidates.end(),
+                [](LogDeleteCandidate const &a, LogDeleteCandidate const &b) { return a.mtime > b.mtime; });
+    });
+
+    while (candidate_count > 0) {
       if (space_to_write(headroom + log_buffer_size)) {
         Debug("logspace", "low water mark reached; stop deleting");
         break;
       }
 
-      Debug("logspace", "auto-deleting %s", candidates[victim].name);
-
-      if (unlink(candidates[victim].name) < 0) {
-        Note("Traffic Server was Unable to auto-delete rolled "
-             "logfile %s: %s.",
-             candidates[victim].name, strerror(errno));
+      // Select the group with biggest ratio
+      auto target =
+        std::max_element(deleting_info.begin(), deleting_info.end(), [](LogDeletingInfo const &A, LogDeletingInfo const &B) {
+          double diff =
```

Comment:
Why store in `diff`? Why subtract at all, instead of returning the relation of the two values?
```
return a.size/a.count < b.size/b.count; // basically.
```